File: test_db.js
const { createClient } = require('@supabase/supabase-js');
const fs = require('fs');
const path = require('path');

const envPath = path.join(process.cwd(), 'frontend', '.env.local');
const env = fs.readFileSync(envPath, 'utf8');
const urlMatch = env.match(/VITE_SUPABASE_URL=(.*)/);
const keyMatch = env.match(/VITE_SUPABASE_ANON_KEY=(.*)/);

const supabase = createClient(urlMatch[1].trim(), keyMatch[1].trim());

async function check() {
  const { data: artists } = await supabase.from('artists').select('*, songs(id, title)');
  if (!artists) return;
  const sorted = artists.sort((a,b) => (b.songs ? b.songs.length : 0) - (a.songs ? a.songs.length : 0));
  console.log("Top artists by song count:");
  for (let i=0; i<3; i++) {
     console.log(sorted[i].name, sorted[i].songs.length);
  }
}

check();

File: test_db.py
import asyncio
from supabase import create_client
import os
from dotenv import load_dotenv

load_dotenv("frontend/.env.local")
supabase = create_client(os.environ["VITE_SUPABASE_URL"], os.environ["VITE_SUPABASE_ANON_KEY"])

response = supabase.table("songs").select("title, artist_id").limit(5).execute()
print(f"Sample songs: {response.data}")

# check an artist
artists = supabase.table("artists").select("id, name").limit(1).execute()
print(f"Sample artist: {artists.data}")

if artists.data:
    aid = artists.data[0]['id']
    s = supabase.table("songs").select("*").eq("artist_id", aid).execute()
    print(f"Songs for {aid}: {len(s.data)}")
